server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -289,6 +294,38 @@ public boolean rename(Path path, Path newPath) throws IOException {
     return source.rename(path, newPath);
   }
 
+  @Override
+  public List<Future<Boolean>> bulkRename(Map<Path,Path> oldToNewPathMap,
+      SimpleThreadPool workerPool, String transactionId) throws InterruptedException {
+    List<Future<Boolean>> results = new ArrayList<>();
+    oldToNewPathMap.forEach((oldPath, newPath) -> results.add(workerPool.submit(() -> {
+      boolean success;
+      try {
+        success = rename(oldPath, newPath);
+      } catch (IOException e) {
+        // The rename could have failed because this is the second time its running (failures
+        // could cause this to run multiple times).
+        if (!exists(newPath) || exists(oldPath)) {
+          throw e;
+        }
+        log.debug(
+            "Ignoring rename exception for {} because destination already exists. orig: {} new: {}",
+            transactionId, oldPath, newPath, e);
+        success = true;
+      }
+      if (!success) {
+        log.error("Rename operation {} returned false. orig: {} new: {}", transactionId, oldPath,
```

Comment:
should this throw an exception?

Reply:
My plan was to preserve current behavior of both methods.  Making this throw an exception would change the behavior where one file failing to rename fails the whole operation.

Reply:
The code that handles the futures deals with execution exceptions and false returns.  Could possibly make this simpler and not return anything `Future<Void>` and only deal with execution exceptions.

Reply:
This does seem to simplify things some.  You still suggesting we throw an error here when ```!success```?

Reply:
I took a quick look at the code calling bulk rename and I think it always fails if any of the futures fail.   I was wondering if it would make sense to move the future handling inside bulk rename and just have bulkRename throw an IOE if any future fails.  As long as this can still result in good error messages to users, I think may simplify the code calling bulk rename alot.

Reply:
This did seem to clean up the handling in the FATE methods.  Checkout my latest, I think this is ready to go now.